packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -189,7 +189,6 @@ export function entityDeserializer(edmToTs, extractODataETag) {
   function deserializeComplexType<
     ComplexTypeNamespaceT extends ComplexTypeNamespace
   >(json: MapType<any>, complexType: ComplexTypeNamespaceT): any {
-    validateComplexTypeJson(json, complexType);
```

Comment:
this means, instead of throwing an error when handling invalid property name, we ignore it?

Reply:
Yes, because we do this when deserializing anyways (we also have a test that checks that this is ignored). Before the behavior was different depending on which function you call, although it should have the same functionality